### ISSUE TITLE
S3 export / COPY statement not always working 

### ISSUE THREAD

**Issue by Mugiwara84** [duckdb/duckdb#7167]
### What happens?

Within the same program (having the same S3 configuration specified), I have no problem doing a S3 import with the HTTPFS extension, regardless of my network configuration.
But I when I do a S3 export, depending on my network configuration, it works or not. For example when I deploy my program on a server which cannot access internet, it does not work. Even though the S3 import is still working.
When it does not work, no log and it seems nothing is done.

I managed to monitor the HTTP network and I see that there are requests that are sent to an aws amazon endpoint when I run a S3 export. And the S3 import is still working with the S3 I specified.

So I am confused !

Same problem using Java or Python.

### To Reproduce

My S3/httpfs configuration looks like : 
        stmt.execute("INSTALL 'path/httpfs.duckdb_extension';");
        stmt.execute("LOAD httpfs;");
        stmt.execute("SET s3_url_style='path';");
        stmt.execute("SET s3_endpoint='my-minio.endpoint.fr';");
        stmt.execute("SET s3_access_key_id='" + S3_ACCESS_KEY_ID + "';");
        stmt.execute("SET s3_secret_access_key='" + S3_SECRET_ACCESS_KEY + "';");

I'm using those kind of statements for S3 import and S3 export :

select count(*) FROM parquet_scan('s3://mybucket/myparquet.parquet')

COPY (SELECT '42') TO 's3://mybucket/42.csv'

The first statement works fine. The second one seems to fail depending on my network config.

### OS:

Windows and Linux

### DuckDB Version:

0.7.1

### DuckDB Client:

Java and Python

### Full Name:

Guillaume Jardillier

### Affiliation:

Insee

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mugiwara84** [comment]
Hi @samansmink,
my description wasn't clear, I apologize.

I did many different tests.
On my professional computer, when I am at my workplace, I can do import and export on an internal MinIO repository using DuckDB.
On my professional computer, when I am at home, I can only do import, export does not work, the process is running, never stops, and seems to do nothing.
I use exactly the same code but network policy/internet access isn't exactly the same depending on my location (work or home). I do not know the specifics unfortunately.

When I deploy my program on a server in an internal network : which can access my internal minio repository but cannot access an external website :
- export fails and we were able to monitor some requests to an amazon aws endpoint : s3-1-w.amazonaws.com
- import works and uses my minio repository : we do not monitor requests to amazon
Still the same code with the same MinIO configuration. So I am very surprised to monitor those requests to amazon even though my endpoint is correctly set since the import is working.
I hope it is more clear !
